photon-lib/src/main/scala/com/linkedin/photon/ml/hyperparameter/search/GaussianProcessSearch.scala

```diff
@@ -116,22 +132,65 @@ class GaussianProcessSearch[T](
   }
 
   /**
-   * Handler callback for each observation. In this case, we record the observed point and values.
+   * Handler for adding new points and evaluation values
    *
-   * @param point the observed point in the space
-   * @param eval the observed value
+   * @param pastPoints the past set of points
+   * @param pastEvals the past current set of evaluation values
+   * @param pastBestEval the past best value
+   * @param point the new point to be added
+   * @param eval the new evaluation value
+   *
+   * @return the new set of points, evaluations and best evaluation
    */
-  protected[search] override def onObservation(point: DenseVector[Double], eval: Double): Unit = {
-    observedPoints = observedPoints
+  protected[search] def addObservation(
+      pastPoints: Option[DenseMatrix[Double]],
+      pastEvals: Option[DenseVector[Double]],
+      pastBestEval: Double,
+      point: DenseVector[Double],
+      eval: Double): (Option[DenseMatrix[Double]], Option[DenseVector[Double]], Double) = {
+
+    val newPoints = pastPoints
       .map(DenseMatrix.vertcat(_, point.toDenseMatrix))
       .orElse(Some(point.toDenseMatrix))
 
-    observedEvals = observedEvals
+    val newEvals = pastEvals
       .map(DenseVector.vertcat(_, DenseVector(eval)))
       .orElse(Some(DenseVector(eval)))
 
-    if (evaluator.betterThan(eval, bestEval)) {
-      bestEval = eval
+    val newBest = if (evaluator.betterThan(eval, pastBestEval)) { eval } else { pastBestEval }
+
+    (newPoints, newEvals, newBest)
+  }
+
+  /**
+   * Handler callback for each observation. In this case, we record the observed point and values.
+   *
+   * @param point the observed point in the space
+   * @param eval the observed value
+   */
+  protected[search] override def onObservation(
+      point: DenseVector[Double],
+      eval: Double,
+      priorData: Boolean = false): Unit = {
+
+    if(priorData) {
```

Comment:
If `priorData` is true only within this class, it would be better to handle this with a separate `onPriorObservation` function, IMO.

Reply:
Thanks! I'll do the change.